registry/remote/referrers.go

```diff
@@ -68,6 +68,32 @@ var (
 	errNoReferrerUpdate = errors.New("no referrer update")
 )
 
+const (
+	// OpDeleteReferrerIndex represents the operation for deleting a
+	// referrer index.
+	OpDeleteReferrerIndex = "DeleteReferrersIndex"
```

Comment:
How about adding a new method to `ReferrersError` like
```go
func (e *ReferrersError) IndexDeleteNotAllowed() bool {
    if e.Op != OpDeleteReferrerIndex {
        return false
    }

    var errResp *errcode.ErrorResponse
    if errors.As(e.Err, &errResp) {
        return errResp.StatusCode == http.StatusMethodNotAllowed 
    }
    return false
}
```
so that we can make `OpDeleteReferrerIndex` private.

Also, easier to use
```golang
err = repo.Push(ctx, referrerDescriptor, bytes.NewReader(referrerManifestContent))
if err != nil {
    var re *remote.ReferrersError
    if !(errors.As(err, &re) && re.IndexDeleteNotAllowed()) {
        return err
    }
    // the push is in a degraded status where index delete is not allowed
}
```

Reply:
+1 on making the operation string private, but I don't think we should add `http.StatusMethodNotAllowed` into the wrapper function. The return code can be Forbidden or anything else.

Reply:
changed according with caller side code updated in the draft PR: https://github.com/oras-project/oras/pull/921